File: LeetCode/Graph/Euler/python/Leetcode565-array-nesting.py
class Solution:
    def arrayNesting(self, nums: List[int]) -> int:
        n,ans = len(nums),0
        for i in range(n):
            if nums[i] == -1:  continue
            cur,ne = 0,i
            while nums[ne] != -1:
                cur += 1
                t = ne
                ne = nums[ne]
                nums[t] = -1
            ans = max(cur,ans)
        return ans

File: LeetCode/Graph/Euler/Rust/Leetcode565-array-nesting.rs
impl Solution {
  pub fn array_nesting(nums: Vec<i32>) -> i32 {
    let (mut ans, n) = (0, nums.len());
    let mut st = vec![false; n];
    for i in 0..n {
      if st[i] {
        continue;
      }
      let (mut ne, mut cur) = (i, 0);
      while !st[ne] {
        cur += 1;
        st[ne] = true;
        ne = nums[ne] as usize;
      }
      ans = ans.max(cur);
    }
    ans
  }
}
